org.eclipse.scanning.server/src/org/eclipse/scanning/server/servlet/ScanProcess.java

```diff
@@ -209,6 +175,61 @@ public void execute() throws EventException {
 		}
 	}
 
+	private void executeNonBlocking(IDeviceController controller, ScanBean bean) throws ScanningException, InterruptedException, TimeoutException, ExecutionException {
+		
+		logger.debug("Running non-blocking device {}", controller.getDevice().getName());
+		controller.getDevice().start(null);
+		
+		long latchTime = Long.getLong("org.eclipse.scanning.server.servlet.asynchWaitTime", 500);
+		logger.debug("Latching on device {} for {}", controller.getDevice().getName(), latchTime);
+		controller.getDevice().latch(latchTime, TimeUnit.MILLISECONDS); // Wait for it to do a bit in case of errors.
+		
+		logger.warn("Cannot run end script when scan is async. (Scan has not been cancelled, after script has been ignored.)");
+		logger.warn("Cannot perform end position when scan is async. (Scan has not been cancelled, end has been ignored.)");
+	}
+
+	private void executeBlocking(IDeviceController controller, ScanBean bean) throws ScanningException, InterruptedException, TimeoutException, ExecutionException, EventException, UnsupportedLanguageException, ScriptExecutionException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)